### ISSUE TITLE
Issue in Updating Object Permission on System Administrator Clone Profile by using Metadata API

### ISSUE THREAD

**Issue by deekshaverma26** [financialforcedev/apex-mdapi#254]
**Issue Scenario :**
When I am trying to update object permission after cloning system administrator profile using Metadata API, the object permission is not being updated.

**Working Scenario :** 
If I am trying to update object permission from the user interface first, then updating permission from the Metadata API, then it is working fine.

**Question :** 
Why Metadata API is not updating object permission without updating from user interface first?

**Comment by BoonPlus** [comment]
@deekshaverma26 how do you identify the "System Administrator" profile in your code? If you use the profile name, it is "Admin" for System Administrator.

**Comment by deekshaverma26** [comment]
We are not updating System Administrator profile. I am updating system administrator's cloned profile. So there is no need to pass profile name as 'Admin' . Others profiles are updated successfully but we are facing problem while updating System administrator's cloned profile

**Comment by BoonPlus** [comment]
@deekshaverma26 I've done it for my cloned system admin profile w/o running into any issues. Is there a specially character in the profile name? e.g. "Cloned: System Administrator", ":" should be "%3A" for the profile name in your code.

**Comment by deekshaverma26** [comment]
There is no special character in my cloned profile. The Name is "AdminCloned", now i am trying to update object permission by using metadata api. 

Question : - why it is not updating object permission?

**Comment by BoonPlus** [comment]
@deekshaverma26 Could you share the code?

**Comment by BoonPlus** [comment]
@deekshaverma26 another question, does the code work for a different profile?

**Comment by deekshaverma26** [comment]
Yes , code is working for different profile fine. As well as i am getting result success as true for system administrator cloned profile but When i checking object permission from the user interface then there is no changes showing.

**Comment by deekshaverma26** [comment]
There is one thing i mentioned in earlier comment that is following:--

Issue Scenario :
When I am trying to update object permission after cloning system administrator profile using Metadata API, the object permission is not being updated.

Working Scenario :
If I am trying to update object permission from the user interface first, then updating permission from the Metadata API, then it is working fine.

did you understand my scenario because I need solution asap ?

**Comment by BoonPlus** [comment]
I ran into an issue once that some profiles need to be touched first (e.g. touch = update the profile description through the user interface and save) before the permission can be updated from the Metadata API. Maybe you can to try to execute your profile upsertMetadata code twice and see if see it makes any difference?

**Comment by deekshaverma26** [comment]
So did you get this issue or not ?

**Comment by Rajith26** [comment]
Hi Deeksha, Were you able to make this change through mdapi?

**Comment by Rajith26** [comment]
> So did you get this issue or not ?

Hi Deeksha, Were you able to make this change through mdapi?
